### PR TITLE
Remove support for PHP 8.0, Upgrade PHPUnit to 10.x

### PR DESCRIPTION
**Pull request by gsteel**
Closes #100

- Bumps Service Manager to ^3.21.0 for PHPUnit 10 Compat
- Drops support for PHP 8.0
- Upgrades PHPUnit to 10.1.x

### CODE REVIEW

**Review by Ocramius — APPROVED** [review]
Thanks @gsteel!

**Comment by Ocramius** [comment]
> snappy_compress : expects parameter to be string.

Nnnnnggghhh

**Comment by gsteel** [comment]
I think I'm gonna have to disable the `rar` extension and tests:
https://github.com/cataphract/php-rar/issues/17

Not sure why it was building a few weeks ago - perhaps a PHP patch release broke it.

**Comment by gsteel** [comment]
Green now @Ocramius 😅

File: test/Compress/SnappyTest.php [hunk 1]
```diff
@@ -40,10 +40,16 @@ public function testBasicUsage(): void
...
+        set_error_handler(function ($num, $message): never {
+            throw new TypeError($message);
+        });
+
+        $this->expectException(TypeError::class);
         $this->expectExceptionMessage('snappy_compress : expects parameter to be string');
         /** @psalm-suppress InvalidArgument, InvalidCast */
         (new SnappyCompression())->compress([]);
+
+        restore_error_handler();
```
Comment (Ocramius) [inline test/Compress/SnappyTest.php]: This will never be reached

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: test/Compress/SnappyTest.php (hunk 1 above)
Comment (gsteel) [reply test/Compress/SnappyTest.php]: 🤦‍♂️

**Review by Ocramius — APPROVED** [review]
Thanks @gsteel

**Pull request merged**